Assets/Scripts/RemoteInput.cs

```diff
@@ -16,14 +18,47 @@ enum EventType
         Mouse = 1,
         MouseWheel = 2,
         Touch = 3,
-        ButtonClick = 4
+        ButtonClick = 4,
+        Gamepad = 5,
+    }
+
+    enum GamepadEventType {
+      ButtonUp = 0,
+      ButtonDown = 1,
+      ButtonPressed = 2,
+      Axis = 3
+    }
+
+    enum GamepadKeyCode {
+        Button0 = 0,
+        Button1,
+        Button2,
+        Button3,
+        Button4,
+        Button5,
+        Button6,
+        Button7,
+        Button8,
+        button9,
+        axis0Button,
+        axis1Button,
+        dpadUp,
+        dpadDown,
+        dpadLeft,
+        dpadRight,
+        axis0 = 100,
+        axis1
+        
     }
 
     public static class RemoteInput
     {
         public static Keyboard Keyboard { get; private set; }
         public static Mouse RemoteMouse { get; private set; }
         public static Touchscreen RemoteTouch { get; private set; }
+
+        static Dictionary<double, GamepadState> GamepadsStates;
```

Comment:
Why is this using `double` type for id?
You can use `BitConverter.ToInt64` or `BitConverter.ToSingle`.

Reply:
Because in the javascript side we are using the timestamp that the gamepad was connected to the client machine to identity each unique gamepad.

Reply:
We expect using `short` data type.

Provide the ID from Unity side.
Initialize process of the `DataChannel` has

Reply:
`byte` instead of `short`

Reply:
@karasusan @sindharta Just think about the idea of generate global IDs for gamepads in unity side and there's a problem: 
Now we only have a single-direction data channel between unity and web browser. We could only send data via `sendMsg` function from javascript side to unity side. So that means unity can generate IDs but there's no way to let web browsers know. 
That would be another topic if we want to make it possible to send messages from unity to javascript.